src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -996,6 +992,32 @@ public int Count
         }
 
         /// <summary>
+        /// Gets the number of key/value pairs contained in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>. Should only be used after all locks
+        /// have been aquired.
+        /// </summary>
+        /// <exception cref="T:System.OverflowException">The dictionary contains too many
+        /// elements.</exception>
+        /// <value>The number of key/value pairs contained in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>.</value>
+        /// <remarks>Count has snapshot semantics and represents the number of items in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>
+        /// at the moment when Count was accessed.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int GetCountInternal()
+        {
+            int count = 0;
```

Comment:
It might be well to assert the locks are held here. Maybe with something like:
```C#
[Conditional("DEBUG")]
private void AssertAllLocksHeld()
{
    object[] locks = _tables._locks;
    for (int i = 0; i < locks.Length; i++)
    {
        Debug.Assert(Monitor.IsEntered(locks[i]), $"Expected lock {i} is not held.");
    }
}